plugins/inputs/cloud_pubsub/pubsub.go

```diff
@@ -119,13 +127,46 @@ func (ps *PubSub) Stop() {
 	ps.wg.Wait()
 }
 
-func (ps *PubSub) subReceive(cctx context.Context) {
+// startReceiver is called within a goroutine and manages keeping a
+// subscription.Receive() up and running while the plugin has not been stopped.
+func (ps *PubSub) receiveWithRetry() {
+	err := ps.startReceiver()
+
+	for err != nil && ps.ctx.Err() == nil {
+		log.Printf("E! Receiver for subscription %s exited with error: %v", ps.sub.ID(), err)
```

Comment:
On all the logging calls, start them with: `E! [inputs.cloud_pubsub]` (or `I!`, `D!`).  We are looking to simplify this in the future but for now this helps keep straight where the log message is coming from.